### ISSUE TITLE
ReplicaSets deployed by Argo don't show their pods

### ISSUE THREAD

**Issue by dbason** [rancher/rancher#31286]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Deploy a ReplicaSet with the following YAML
```
apiVersion: apps/v1
kind: ReplicaSet
metadata:
  annotations:
    rollout.argoproj.io/desired-replicas: "3"
    rollout.argoproj.io/revision: "3"
  labels:
    app: argo-demo-rollouts
    rollouts-pod-template-hash: 6b4d44785c
  name: argo-demo-rollouts-6b4d44785c
  namespace: default
  ownerReferences:
  - apiVersion: argoproj.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Rollout
    name: argo-demo-rollouts
    uid: e64580d7-a1fe-4735-8127-f55250a04c88
spec:
  replicas: 3
  selector:
    matchLabels:
      app: argo-demo-rollouts
      rollouts-pod-template-hash: 6b4d44785c
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: argo-demo-rollouts
        rollouts-pod-template-hash: 6b4d44785c
    spec:
      containers:
      - image: ubuntu
        imagePullPolicy: IfNotPresent
        name: argo-demo-rollouts
        ports:
        - containerPort: 80
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
```

**Result:**
ReplicaSet doesn't show any pods in the group by workload page in the Cluster Manager
The ReplicaSet doesn't show up in the Overview section in the Cluster Overview

**Other details that may be helpful:**
This is related to https://github.com/rancher/rancher/issues/27923. The reproduction needs to include the Argo ownerReferences section.  In the pods the workload shows up as a Rollout.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.3
- Installation option (single install/HA): HA

gz#14993
gz#13893

**Comment by vincent99** [comment]
For review/QA: the intended behavior is to hide workloads from the overview page if they are "owned" by a higher-level workload type. For example a Deployment creates ReplicaSets which create Pods. We only want to see the one deployment on the Overview, not the n ReplicaSets and m Pods associated with it.

<hr/> 

On the new/dashboard/explorer/**Vue** side, this filtering is done in the UI:

The previous behavior implemented that as "hide anything that has any ownerReference of any kind". Which does mostly work. But Argo has a controller which creates ReplicaSets and marks itself as the owner . So we were previously hiding these entries because they have some owner, but want to show them by now checking if (any of the) owner(s, there can be multiple) are another Workload type.

The ownerRef Argo puts in is NOT to a Workload type, so the end result is that owner gets ignored and the workload will be shown. In the normal Deployment->ReplicaSet->Pod case, the ReplicaSet will have the Deployment as the owner, which is a Workload type, and so the ReplicaSet will (still) be hidden.

<hr/> 

On the old/manager/**Ember** side, this filtering is done by the API, so the other PR is to rancher/rancher and causes that entry to be returned in the list workloads call so the UI will render it.

**Comment by slickwarren** [comment]
linking https://github.com/rancher/dashboard/pull/2376/files for master dashboard updates

**Comment by slickwarren** [comment]
I was able to test the following:
***Dashboard***
* tested on v2.5.6 and master-head (8ba83dc) using 1 downstream cluster
using the yaml from the original post in this thread, I was able to confirm that the replicaset that is "owned" by a higher-level workload type is displayed in `overview` for workloads
<img width="933" alt="Screen Shot 2021-03-11 at 12 39 55 PM" src="https://user-images.githubusercontent.com/16691014/110852361-1fb30380-8267-11eb-815d-aa26fe36d9db.png">

* I also added a replicaset with a custom owner, that also shows up properly in Dashboard

***Ember UI***
* tested on master-head (8ba83dc) using 1 downstream cluster
using the yaml from the original post in this thread, I was able to confirm the same case as above for Ember UI.  Below, I show that pods are now displayed both in the deployment and on the workload overview page.
<img width="1312" alt="Screen Shot 2021-03-11 at 12 44 31 PM" src="https://user-images.githubusercontent.com/16691014/110852729-918b4d00-8267-11eb-810c-387448824c4b.png">

<img width="536" alt="Screen Shot 2021-03-11 at 12 44 38 PM" src="https://user-images.githubusercontent.com/16691014/110852721-8fc18980-8267-11eb-8db0-6698b9fd8e4e.png">

* I also added a replicaset with a custom owner, that also shows up properly in Ember

### Why this is reopened
The ember UI has not been fixed on 2.5-head as of (a420b5195dd1d0a23411b0aeb17713f291e90e10).  Meaning that there are no associated pods for these types of workloads
<img width="1359" alt="Screen Shot 2021-03-11 at 2 59 14 PM" src="https://user-images.githubusercontent.com/16691014/110866682-5e9e8480-827a-11eb-8c5a-55a78b862f36.png">
<img width="1335" alt="Screen Shot 2021-03-11 at 2 59 07 PM" src="https://user-images.githubusercontent.com/16691014/110866687-5fcfb180-827a-11eb-9417-2989d60a8484.png">

**Comment by dramich** [comment]
@slickwarren This has already been in fixed in the 2.5 line, you are running a hash that is 3 days old and is from before the merge of the fix. `head` should be using the latest hash.

**Comment by slickwarren** [comment]
I validated the exact steps above on 2.5-head (38649e228b183289ddf4b876c3ca5dc1615644cc), with the additional tests listed below:
* added all deployment types -- dashboard shows overview as expected from  https://github.com/rancher/rancher/issues/31286#issuecomment-780919645
* updated pod count in workload for daemonset that is  "owned" by a higher-level workload type -- updated and displayed properly
